### PR TITLE
Adds GET Support, Switch to Official Python Docker Image, and other goodness

### LINKED ISSUE #3: Add GET http method to parsing microservice

### PR DESCRIPTION
**Pull request by hkeeler**
Closes #3.

This PR was originally intended to just add `GET` support to the API, but then turned into a _bit_ more:
- Adds `method` param for specifying whether to use usaddress's `parse` or `tag` function
- Adds `validate` param to check if address string is either incomplete or contains invalid address parts, like P.O. Box
- Adds `time` and `host` attributes to `/status` resource
- Fixes error handling bug
- Adds InvalidApiUsage exception and error handling
- Adds PyDoc strings to all functions

In addition, I've switched the `Dockerfile` around to use the [Official Python Docker Image](https://registry.hub.docker.com/_/python/).  This makes the Docker setup super clean, and automatically installs all Python dependencies for you so long as you put a `requirements.txt` in the root of your project.

### CODE REVIEW

**Comment by hkeeler** [comment]
Just realized I haven't updated the README to reflect these changes.  Let me get those in before anyone merges this.

File: Dockerfile [hunk 1]
```diff
@@ -2,21 +2,9 @@
 # To build, run docker build --rm --tag=hmda/grasshopper-parser .
 # A container can be started by running docker run -ti -p 5000:5000 hmda/grasshopper-parser
 
-FROM ubuntu:latest
+FROM python:2.7.9-onbuild
 MAINTAINER Juan Marin Otero <juan.marin.otero@gmail.com>
```
Comment (jmarin) [inline Dockerfile]: @hkeeler You should be the maintainer of this now! :)

Comment (hkeeler) [inline Dockerfile]: I was actually thinking about what/who we should put in the `MAINTAINER` field for all these projects.  I _think_ we want it to be the same across all of our images...and it sounds like it makes better use Docker's cache if its the same too:
- http://crosbymichael.com/dockerfile-best-practices.html

Perhaps a hmdaops-docker@cfpb.gov address at some point?

For now, I'll switch if over to me. :)

**Comment by hkeeler** [comment]
Done tweaking with the README for now.  Feel free to review, merge, etc.

File: README.md [hunk 2]
```diff
@@ -1,54 +1,117 @@
...
+
+    Values:
+    * `false` (Default)
+    * `true`
+
+
+#### Request
+
+    GET http://localhost:5000/parse\?address\=1311+30th+st+washington+dc+20007
+
```
Comment (jmarin) [inline README.md:92]: Shouldn't this be http://localhost:5000/parse?address=1311+30th+st+washington+dc+20007?

File: README.md [hunk 3]
```diff
@@ -1,54 +1,117 @@
...
+JSON attributes via `HTTP POST`.
+
+* **`address`:** Free-text address string to be parsed.
+
+* **`method`:** (Optional) [usaddress parsing method](http://usaddress.readthedocs.org/en/latest/#usage)
+    to be used to split `address` into its component parts.
+
+    Values:
+    * `parse` (Default)
+    * `tag`
```
Comment (jmarin) [inline README.md:78]: Would it be clearer to have a separate /tag URL for this? When I try adding &method=tag to the URL I get the same results, which I guess is to be expected but it also might be confusing

File: app/app.py [hunk 4]
```diff
@@ -1,36 +1,184 @@
...
+        Exception.__init__(self)
+        self.message = message
+        if status_code is not None:
+            self.status_code = status_code
+
+
 app = Flask(__name__)
 
+@app.route('/explode', methods=['GET'])
+def explode():
```
Comment (jmarin) [inline app/app.py:101]: I love the idea of an explode method :) 
But what is this supposed to do? Is this to test something that is coming later?

Comment (hkeeler) [inline app/app.py:101]: I added it to simulate a 500 system error for testing the [`default_error`](https://github.com/cfpb/grasshopper-parser/pull/4/files#diff-938ace9e4d2736ba3e32818c31d7939eR176) error handler.  It works. :)

Not sure it has much long-term usefulness.  I think we can remove it and find a smarter way to simulate a 500 when we tackle the automated testing issue #2.

File: README.md (hunk 3 above)
Comment (hkeeler) [inline README.md:78]: I'm not sure the `method` param will stick around long-term.  I'm guessing we'll figure out which one works the best for us, and stick with that.  I think `/parse` if the better resource name whichever method we go with long term, so I think I'd rather stick with it and keep the params to pick which method we want.

For some examples of addresses that parse differently with `parse` vs. `tag`, they give the following in the [usage](http://usaddress.readthedocs.org/en/latest/#usage) docs.
- `Robie House, 5757 South Woodlawn Avenue, Chicago, IL 60637`
- `State & Lake, Chicago` - don't forget to escape the amperstand with `%26` on this one.
- `P.O. Box 123, Chicago, IL`

You'll notice things like place names get merged into a single `BuildingName` attribute and commas are stripped when using `tag`.

File: README.md (hunk 2 above)
Comment (hkeeler) [inline README.md:92]: Yeah, looks like I left some of the escape characters in there from `curl`.  I'll clean that up.

**Pull request merged**